src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -1719,6 +1765,13 @@ def __init__(
             if query_id_place_holder
             else f"query_id_place_holder_{generate_random_alphanumeric()}"
         )
+        # This is a temporary workaround to handle the case when a snowflake plan is created
```

Comment:
i am not sure if that is a temporary workaround. The sessions scoped temp table could be a long term solution, but it may have other draw back also, you can mention this as an alternative solution below